### ISSUE TITLE
Feature: be more forgiving with format of .node-version file

### ISSUE THREAD

**Issue by shadowspawn** [tkareine/chnode#5]
Disclaimer: I am not a `chnode` user, so feel free to close this issue!

I have been accumulating information on utilities that support use of `.node-version` file and have added `chnode`: https://github.com/shadowspawn/node-version-usage

You might be interested in supporting `.node-version` file with:
- no EOL
- Windows EOL
- leading v on version

**Comment by shadowspawn** [comment]
I experimented with modifying `auto.sh` like this, and worked for my manual tests:

```bash
chnode_auto() {
    local dir="$PWD/" version

    until [[ -z $dir ]]; do
        dir=${dir%/*}

        local version
        if [[ -e "$dir/$CHNODE_AUTO_VERSION_FILENAME" ]]; then
            # read returns a non-zero status but does still work if there is no line ending
            <"$dir/$CHNODE_AUTO_VERSION_FILENAME" read -r version 2>/dev/null
            if [[ -n ${version:-} ]]; then
                # trim possible trailing \d from a Windows created file
                version="${version%%[[:space:]]}"
                # trim possible leading v
                version="${version#v}"
                if [[ $version == "${CHNODE_AUTO_VERSION:-}" ]]; then
                    return
                else
                    CHNODE_AUTO_VERSION=$version
                    chnode "$version"
                    return $?
                fi
            fi
        fi
    done

    if [[ -n ${CHNODE_AUTO_VERSION:-} ]]; then
        chnode_reset
        unset CHNODE_AUTO_VERSION
    fi
}
```

**Comment by tkareine** [comment]
I'll look into this. The work is in the [more-forgiving-parser-for-node-version-file](https://github.com/tkareine/chnode/tree/more-forgiving-parser-for-node-version-file) branch.

**Comment by shadowspawn** [comment]
An apology that my sample code blindly deleted the leading `v`. I was testing with cached node versions that were named with plain semver, and in this context didn't realise there might be other patterns for the read version.

That complicates the approach since if the cached node versions are in folders that _do_ start with a leading `v`, don't want to delete it here! Being more flexible about the prefix is a harder change than coping with more line endings.

**Comment by tkareine** [comment]
No worries – your sample code gave me a good start. I think trimming the leading `v` character away would be okay if a digit follows it. I rewrote the specification for the supported format of the `.node-version` file in the [README](https://github.com/tkareine/chnode/blob/more-forgiving-parser-for-node-version-file/README.md#supported-node-version-file-format). What do you think?

**Comment by shadowspawn** [comment]
Looks good to me. 👍

**Comment by tkareine** [comment]
Ok! Merged and released v0.4.1. Thanks!

**Comment by shadowspawn** [comment]
Nice! Thanks.
